erizo/src/erizo/IceConnection.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * NiceConnection.h
+ */
+
+#ifndef ERIZO_SRC_ERIZO_ICECONNECTION_H_
+#define ERIZO_SRC_ERIZO_ICECONNECTION_H_
+
+#include <boost/scoped_ptr.hpp>
+#include <boost/thread.hpp>
+#include <string>
+#include <vector>
+#include <queue>
+#include <map>
+
+#include "./MediaDefinitions.h"
+#include "./SdpInfo.h"
+#include "./logger.h"
+#include "lib/LibNiceInterface.h"
+
+typedef struct _NiceAgent NiceAgent;
+typedef struct _GMainContext GMainContext;
+typedef struct _GMainLoop GMainLoop;
+
+typedef unsigned int uint;
+
+namespace erizo {
+
+// forward declarations
+typedef std::shared_ptr<dataPacket> packetPtr;
+class CandidateInfo;
+class WebRtcConnection;
+class IceConnection;
+
+struct CandidatePair{
+  std::string erizoCandidateIp;
+  int erizoCandidatePort;
+  std::string clientCandidateIp;
+  int clientCandidatePort;
+  std::string erizoHostType;
+  std::string clientHostType;
+};
+
+class IceConfig {
```

Comment:
LGMT, at some point (not in this PR) we should probably refactor how we pass the configuration. Right now it comes directly from erizoAPI and is filled in progressively in several steps, there's probably a more consistent way to do it.